### ISSUE TITLE
Clean up, optimize, and test Layout.js

### ISSUE THREAD

**Issue by ivebencrazy** [Blanket-Warriors/BentoTime#7]
# Background

Layout.js is right now, kind of crazy logic, mostly stemming from a misuse of RxJS.  It makes it really hard to follow our logic and debug.  Good challenge for anyone who wants to look more into RxJs.  Also, https://github.com/Blanket-Warriors/BentoTime/issues/16 seems to be related to this.

# Needed
 - [x] Clean up Layout.js
 - [x] Write tests for Layout.js

**Comment by mistermjtek** [comment]
:thumbsup: from me! :grin:

**Comment by ivebencrazy** [comment]
Addressed in https://github.com/Blanket-Warriors/BentoTime/pull/18
